### PR TITLE
[INLONG-10012][Agent] Adjust task encapsulation to place common logicin the base class

### LINKED ISSUE #10012: [Improve][Agent] Adjust task encapsulation to place common logic in the base class
### Description

Adjust task encapsulation to place common logic in the base class

### InLong Component

InLong Agent

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by justinwwhuang**
[INLONG-10012][Agent] Adjust task encapsulation to place common logicin the base class
- Fixes #10012

### Motivation

Adjust task encapsulation to place common logicin the base class

### Modifications

Put the basic initialization logic, release logic, and usage of instanceManager into the base class

### Verifying this change

*(Please pick either of the following options)*

- [x] This change is a trivial rework/code cleanup without any test coverage.

- [ ] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

No doc needed

### CODE REVIEW

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/task/AbstractTask.java [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+        } catch (Exception e) {
+            LOGGER.error("start instance manager error: ", e);
+        }
+        initTask();
+        initOK = true;
+    }
+
+    protected abstract void initTask();
+
+    protected abstract void releaseTask();
```
Comment (aloyszhang) [inline inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/task/AbstractTask.java:70]: Add a default empty implementation here and override in the subclass if needed.

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/task/KafkaTask.java [hunk 2]
```diff
@@ -20,86 +20,48 @@
...
-    public String getTaskId() {
-        if (taskProfile == null) {
-            return "";
+    protected List<InstanceProfile> getNewInstanceList() {
+        List<InstanceProfile> list = new ArrayList<>();
+        if (isAdded) {
+            return list;
         }
-        return taskProfile.getTaskId();
+        String dataTime = LocalDateTime.now().format(DateTimeFormatter.ofPattern("yyyyMMddHH"));
```
Comment (aloyszhang) [inline inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/task/KafkaTask.java:58]: Not need to initiate a new DateTimeFormatter every time this method is invoked.

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/task/AbstractTask.java [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+    }
+
+    @Override
+    public void addCallbacks() {
+
+    }
+
+    @Override
+    public void run() {
+        Thread.currentThread().setName("task-core-" + getTaskId());
```
Comment (aloyszhang) [inline inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/task/AbstractTask.java:101]: Better to distinguish the thread name for different types of tasks.

Comment (justinwwhuang) [reply inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/task/AbstractTask.java:101]: At first, I thought the same way, but in reality, different data sources will not be enabled simultaneously in the same process

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/task/AbstractTask.java (hunk 1 above)
Comment (justinwwhuang) [reply inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/task/AbstractTask.java:70]: Yes, some scenarios of releaseTask do not require implementation

**Review by aloyszhang — APPROVED** [review]

**Review by haifxu — APPROVED** [review]

**Review by dockerzhang — APPROVED** [review]

**Pull request merged**